ecl/hql/hqlexpr.cpp

```diff
@@ -5362,29 +5354,12 @@ inline void addActiveTable(UsedExpressionHashTable & array, IHqlExpression * ds)
 CUsedTables::CUsedTables()
 {
     tables.single = NULL;
-    numTables = 0;
     numActiveTables = 0;
 }
 
-CUsedTables::~CUsedTables()
-{
-    if (numTables == 1)
-    {
-        if (numActiveTables == 0)
-            tables.single->Release();
-    }
-    else if (numTables != 0)
-    {
-        for (unsigned i=numActiveTables; i < numTables; i++)
-            tables.multi[i]->Release();
-        delete [] tables.multi;
```

Comment:
@ghalliday Possible memory leak: your probably still need the CUsedTables destructor to delete tables.multi?

Reply:
no - it is correct.  The records that need to be released are no longer added.